lightningd/peer_control.c

```diff
@@ -1611,6 +1613,75 @@ static const struct json_command listpeers_command = {
 /* Comment added to satisfice AUTODATA */
 AUTODATA(json_command, &listpeers_command);
 
+
+static void json_add_peerchannels(struct lightningd *ld,
+                          struct json_stream *response,
+                          struct peer *peer,
+			  enum channel_state *state,
+                          const enum log_level *ll)
+{
+	struct channel *channel;
+
+        if (!state)
+	        json_add_uncommitted_channel(response, peer->uncommitted_channel);
+
+        list_for_each(&peer->channels, channel, list) {
+		if (!state || channel->state == *state) {
+	                if (channel_unsaved(channel))
+        	                json_add_unsaved_channel(response, channel);
+                	else
+                        	json_add_channel(ld, response, NULL, channel);
+		}
+        }
+
+        if (ll)
+                json_add_log(response, ld->log_book, &peer->id, *ll);
+}
+
+static struct command_result *json_listpeerchannels(struct command *cmd,
+						    const char *buffer,
+						    const jsmntok_t *obj UNNEEDED,
+						    const jsmntok_t *params)
+{
+	enum log_level *ll;
+	struct node_id *peer_id;
+	enum channel_state *state;
+	struct peer *peer;
+	struct json_stream *response;
+
+	if (!param(cmd, buffer, params,
+		   p_opt("id", param_node_id, &peer_id),
+		   p_opt("state", param_channel_state, &state),
```

Comment:
I like to maintain the same convention for all the parameters, when we filter by some status, we use the word "status" instead of "state". To avoid confusion for the user.